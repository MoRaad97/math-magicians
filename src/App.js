import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Calculator from './Components/calculator';
import Home from './Components/home';
import Navbar from './Components/navbar';
import Quote from './Components/quote';

const App = () => (
  <BrowserRouter>
    <Navbar />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="calculator" element={<Calculator />} />
      <Route path="quote" element={<Quote />} />
      <Route path="*" element={<div style={{ color: 'red', fontSize: '50px' }}>ERROR PAGE NOT EXIST</div>} />
    </Routes>
  </BrowserRouter>

);

export default App;
