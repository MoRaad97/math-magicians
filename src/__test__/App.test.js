import React from 'react';
// import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../App'

it('render the Calculator Component', ()=> {
const tree = renderer.create(<App />).toJSON();
expect(tree).toMatchSnapshot()
})