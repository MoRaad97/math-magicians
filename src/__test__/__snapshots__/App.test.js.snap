// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Calculator Component 1`] = `
Array [
  <nav>
    <div
      className="nav-title"
    >
      <h1>
        Math-Magicians
      </h1>
    </div>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <section
    className="home"
  >
    <div
      className="container"
    >
      <h2
        style={
          Object {
            "fontSize": "30px",
            "marginTop": "0",
          }
        }
      >
        Welcome To Our Page
      </h2>
      <p
        style={
          Object {
            "fontSize": "25px",
          }
        }
      >
        Mathematics (from Ancient Greek μάθημα; máthēma: knowledge, study, learning) is an area of knowledge that includes such topics as numbers (arithmetic, number theory) formulas and related structures (algebra) shapes and the spaces in which they are contained (geometry) and quantities and their changes.
      </p>
      <p
        style={
          Object {
            "fontSize": "25px",
          }
        }
      >
        This Website has a contents a Calculator which let you do simple calculations and math Quotes.
      </p>
    </div>
  </section>,
]
`;
